week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,71 @@
-function setAlarm() {}
+function setAlarm() {
+  //get the value of the input field
+  let inputField = document.querySelector("#alarmSet").value;
+
+  //set the title to the correct value
+  const timeRemaining = document.querySelector("#timeRemaining");
+  timeRemaining.textContent = `Time Remaining: ${inputField}:00`;
+
+  let myAlarm;
+  ///
+  document.getElementById("set").hidden = true;
+
+  // add `pause` functionality
+  document.getElementById("set").textContent = "Pause";
+
+  //create new button Pause
+  let pauseButton = document.createElement("button");
+  pauseButton.textContent = "Pause";
+  let divButton = document.querySelector(".centre >:last-child");
+  divButton.prepend(pauseButton);
+  pauseButton.className = "pauseButton";
+
+  document.querySelector(".pauseButton").addEventListener("click", () => {
+    clearInterval(myAlarm);
+    document.querySelector(".pauseButton").hidden = true;
+
+    let restartButton = document.createElement("button");
+    restartButton.textContent = "Restart";
+    let divButton = document.querySelector(".centre >:last-child");
+    divButton.prepend(restartButton);
+    restartButton.className = "restartButton";
+
+    //restart functionality
+    document.querySelector(".restartButton").addEventListener("click", () => {
+      location.reload();
+    });
+  });
+
+  //update the `Time Remaining` title every second
+  myAlarm = setInterval(() => {
+    if (inputField == 0) {
```

Comment:
See the commend about validation above. I see you've used `==` rather than `===` to check the equality. This only applies loose equality checking which is why it works in some cases even though the original type of the value is a string. Ideally I would suggest having some function that parses the input field value and converts it to a valid integer or returns some fixed value such as `-1` if it is invalid. This value, if its valid, should be an integer that you can then check to see if it is 0 or not using strict equality (`===`).

Reply:
Thanks for your feedback. I have used a function which checks 2 conditions: the value is positive and contains only digits. Thanks for your assistance.